test/linter/test-versions.ts

```diff
@@ -159,91 +141,103 @@ const checkVersions = (
-            );
-          }
+        if (typeof version === 'string' && version.includes('≤')) {
+          const releaseData =
+            browsers[browser].releases[version.replace('≤', '')];
           if (
-            typeof statement.version_added === 'string' &&
-            typeof statement.version_removed === 'string' &&
-            addedBeforeRemoved(statement) === false
+            !releaseData ||
+            !releaseData.release_date ||
+            new Date(releaseData.release_date) > twoYearsAgo
           ) {
             logger.error(
-              chalk`{bold version_removed: "${statement.version_removed}"} must be greater than {bold version_added: "${statement.version_added}"}`,
+              chalk`{bold ${property}: "${version}"} is {bold NOT} a valid version number for {bold ${browser}}\n    Ranged values are only allowed for browser versions released two years or older (on or before ${twoYearsAgo}). Ranged values are also not allowed for browser versions without a known release date.`,
             );
           }
         }
+      }
 
-        if ('flags' in statement && !browsers[browser].accepts_flags) {
+      if ('version_added' in statement && 'version_removed' in statement) {
+        if (statement.version_added === statement.version_removed) {
           logger.error(
-            chalk`This browser ({bold ${browser}}) does not support flags, so support cannot be behind a flag for this feature.`,
+            chalk`{bold version_added: "${statement.version_added}"} must not be the same as {bold version_removed} for {bold ${browser}}`,
           );
         }
-
-        if (statement.version_added === false) {
-          if (
-            Object.keys(statement).some(
-              (k) => !['version_added', 'notes', 'impl_url'].includes(k),
-            )
-          ) {
-            logger.error(
-              chalk`The data for ({bold ${browser}}) says no support, but contains additional properties that suggest support.`,
-            );
-          }
+        if (
+          typeof statement.version_added === 'string' &&
+          typeof statement.version_removed === 'string' &&
+          addedBeforeRemoved(statement) === false
+        ) {
+          logger.error(
+            chalk`{bold version_removed: "${statement.version_removed}"} must be greater than {bold version_added: "${statement.version_added}"}`,
+          );
         }
+      }
+
+      if ('flags' in statement && !browsers[browser].accepts_flags) {
+        logger.error(
+          chalk`This browser ({bold ${browser}}) does not support flags, so support cannot be behind a flag for this feature.`,
+        );
+      }
 
+      if (statement.version_added === false) {
         if (
-          Array.isArray(supportStatement) &&
-          statement.version_added === false
+          Object.keys(statement).some(
+            (k) => !['version_added', 'notes', 'impl_url'].includes(k),
+          )
```

Comment:
Not new code, but creating an array of keys with `Object.keys()` can be avoided:

```suggestion
          ['version_added', 'notes', 'impl_url'].some((k) => Object.hasOwn(statement, k))
```

Reply:
This actually doesn't work because we're trying to find if there are properties _other_ than those included in the object.  Reverting.